storage/innobase/lock/lock0lock.cc

```diff
@@ -1853,46 +2002,70 @@ lock_rec_lock(
 
   /* Simplified and faster path for the most common cases */
   if (!impl)
-    lock_rec_create_low(nullptr, mode, id, block->page.frame, heap_no, index,
-                        trx, false);
+    lock_rec_create(null_c_lock_info, mode, id, block->page.frame, heap_no,
+                    index, trx, false);
 
   return DB_SUCCESS_LOCKED_REC;
 }
 
-/*********************************************************************//**
-Checks if a waiting record lock request still has to wait in a queue.
+/** Checks if a waiting record lock request still has to wait in a queue.
+@param cell      record locks hash table cell for waiting lock
+@param wait_lock waiting lock
 @return lock that is causing the wait */
-static
-const lock_t*
+static conflicting_lock_info
 lock_rec_has_to_wait_in_queue(const hash_cell_t &cell, const lock_t *wait_lock)
```

Comment:
The `@return` comment should be adjusted.

This function looks very similar to `lock_rec_other_has_conflicting()`. Would it be feasible to unify these somehow, perhaps by defining some `template` parameter?

Reply:
I would not do such changes in the current task, as they are potentially bug-prone, what can require more time for testing, debugging, delaying the release of the solution for unpredictable amount of time.

Reply:
I agree. It could make sense to file a follow-up ticket for that. After some review, we might also end concluding that it is not feasible, for some specific reasons that should be written to the ticket.